feat(UNIX_TIMESTAMP): add basic support for UNIX_TIMESTAMP function

rtm-ctrlz opened the pull request:
Added basic support for `UNIX_TIMESTAMP()` function:

- without arguments: returns current timestamp
- with single argument: transforms string representation of date/datetime to timestamp

Things could be done:
 - add some checks for invalid values/formats
 - add support for timezones
 - support for floating point type of return value
   ```
   mysql> SELECT UNIX_TIMESTAMP('2015-11-13 10:20:19.012');
       -> 1447431619.012
   ```

Docs for function - [MySQL 8.0 Reference Manual](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_unix-timestamp)

sergiosalvatore commented:
Hi -- thanks for the submission!  Would you be able to address the issues Psalm flagged?

rtm-ctrlz commented:
@sergiosalvatore, I'm not sure that I know proper way to do it.

After `$column = Evaluator::evaluate($conn, $scope, $args[0], $row, $result);` line variable `$column` may have almost any type, so I see some ways:
 - convert to string and let `strtotime` handle it;
 - add check for type and pass only string, otherwise return `NULL` like original function do.

Also I'll add some things to this PR a bit later:
 - fix for a typo (exception message)
 - convertion of `false`

rtm-ctrlz commented:
Sorry, misclicked close button

rtm-ctrlz commented:
@sergiosalvatore , have look, things done:

- fixed couple typos
- fixed psalm issue
- fixed return type to `int | null` , like original function
- with invalid values (when `strtotime` returns `false`) function will return `NULL`, like original function
- "refactored" from if's to switch for argument count check

rtm-ctrlz commented:
bump ;)

muglug commented:
Thanks!
